contracts/staking/TokenStaking.sol

```diff
@@ -758,24 +775,32 @@ contract TokenStaking is Ownable, IStaking {
 
         (uint256 keepStakeAmount, , uint256 undelegatedAt) = keepStakingContract
             .getDelegationInfo(operator);
-        (uint96 tAmount, ) = keepToT(keepStakeAmount);
+
+        (uint96 realKeepInTStake, ) = keepToT(keepStakeAmount);
+        uint96 oldKeepInTStake = operatorStruct.keepInTStake;
 
         require(
-            operatorStruct.keepInTStake > tAmount || undelegatedAt != 0,
+            oldKeepInTStake > realKeepInTStake || undelegatedAt != 0,
             "There is no discrepancy"
         );
-        operatorStruct.keepInTStake = tAmount;
+        operatorStruct.keepInTStake = realKeepInTStake;
         seizeKeep(
             operatorStruct,
             operator,
             stakeDiscrepancyPenalty,
             stakeDiscrepancyRewardMultiplier
         );
 
-        uint96 slashedAmount = tAmount - operatorStruct.keepInTStake;
+        uint96 slashedAmount = realKeepInTStake - operatorStruct.keepInTStake;
         emit TokensSeized(operator, slashedAmount, true);
         if (undelegatedAt != 0) {
             operatorStruct.keepInTStake = 0;
+            decreaseStakeCheckpoint(operator, oldKeepInTStake);
+        } else {
+            decreaseStakeCheckpoint(
+                operator,
+                oldKeepInTStake - operatorStruct.keepInTStake
+            );
         }
```

Comment:
```suggestion
        }
        decreaseStakeCheckpoint(
            operator,
            oldKeepInTStake - operatorStruct.keepInTStake
        );
```
I think this will cost less bytes

Reply:
Done